sources/web/datalab/polymer/test/item-list-test.ts

```diff
@@ -500,6 +515,145 @@ describe('<item-list>', () => {
+      const headerIcons = testFixture.$.header.querySelectorAll('.sort-icon');
+      assert(!headerIcons[0].hidden, 'first column should show sort icon');
+      assert(headerIcons[0].icon === 'arrow-upward',
+          'first column should show ascending sort icon');
+      assert(headerIcons[1].hidden, 'second column icon should be hidden');
+
+      testFixture._sortBy(1);
+      testFixture.$.list.render();
+      assert(headerIcons[0].hidden, 'first column icon should be hidden');
+      assert(!headerIcons[1].hidden, 'second column should show sort icon');
+      assert(headerIcons[1].icon === 'arrow-upward',
+          'second column should show ascending sort icon');
+
+      testFixture._sortBy(1);
+      testFixture.$.list.render();
+      assert(headerIcons[0].hidden, 'first column icon should be hidden');
+      assert(!headerIcons[1].hidden, 'second column should show sort icon');
+      assert(headerIcons[1].icon === 'arrow-downward',
+          'second column should show descending sort icon');
+    });
+
+    it('sorts the clicked column', () => {
+      const headerButtons = testFixture.$.header.querySelectorAll('.column-button');
+      const headerIcons = testFixture.$.header.querySelectorAll('.sort-icon');
+      headerButtons[0].click();
+      testFixture.$.list.render();
+      assert(headerIcons[0].icon === 'arrow-downward',
+          'first column should show descending sort icon');
+
+      headerButtons[1].click();
+      testFixture.$.list.render();
+      assert(headerIcons[1].icon === 'arrow-upward',
+          'second column should show ascending sort icon');
+    });
+
+    it('sorts while filtering is active', () => {
+      testFixture.$.filterToggle.click();
+      testFixture._filterString = '*';
+      testFixture.$.list.render();
+
+      let renderedRows = testFixture.$.listContainer.querySelectorAll('.row');
+      // row 0
+      let columns0 = renderedRows[0].querySelectorAll('.column');
+      assert(columns0[0].innerText === testFixture.rows[1].columns[0]);
+      assert(columns0[1].innerText === testFixture.rows[1].columns[1]);
+      // row 1
+      let columns1 = renderedRows[1].querySelectorAll('.column');
+      assert(columns1[0].innerText === testFixture.rows[0].columns[0]);
+      assert(columns1[1].innerText === testFixture.rows[0].columns[1]);
+
+      testFixture._sortBy(0);
+      testFixture.$.list.render();
+      renderedRows = testFixture.$.listContainer.querySelectorAll('.row');
+      // row 0
+      columns0 = renderedRows[0].querySelectorAll('.column');
+      assert(columns0[0].innerText === testFixture.rows[0].columns[0]);
+      assert(columns0[1].innerText === testFixture.rows[0].columns[1]);
+      // row 1
+      columns1 = renderedRows[1].querySelectorAll('.column');
+      assert(columns1[0].innerText === testFixture.rows[1].columns[0]);
+      assert(columns1[1].innerText === testFixture.rows[1].columns[1]);
+    });
+
+    it('sorts numbers correctly', () => {
```

Comment:
Could add a test to see what happens when there are a few values that are equal.

Reply:
Done.